DTS-STN/dts-rmp
# 102 contact form validation

# Description

Adds client side validation and validation visuals to the contact form. Fields will get a red border and an error message below them if you focus and unfocus (blur) the field. On submit, a list of errors will appear at the top of the page, and the browser will automatically scroll to it.

## Test Instructions

1. Navigate to Add contacts & engagements
1. click on any required field, then unselect it. The field should give visual indication that the input in invalid
1. try to submit the form with invalid data (leave required fields blank, type something that isn't an email in a required email field) The page should scroll to the top and display a list of all errors
1. Try to submit the form with VALID data. Make sure it will actually submit. 

## Help Requested

- CSS suggestions would be greatly appreciated. The error list box that appears on invalid data submit is a disaster, but there is no design to follow. If you have CSS experience and are willing to look it over, check it out. 

## Checklist
